main.py

```diff
@@ -51,19 +52,21 @@ class Model(str, Enum):
 
 diffusion_client = None
 diffusion_client_pid = None
+diffusion_client_lock = threading.Lock()
 
 
 @app.get("/generate")
-async def generate(prompt: str, n: int,  model: Model):
+def generate(prompt: str, n: int,  model: Model):
```

Comment:
I replaced `async def` with `def` because `DiffusionClient` doesn't the async interface, thus the blocking call to `client.draw()` can freeze the whole asyncio event loop.

Long story short: FastAPI docs recommend using `def` if a handler calls a heavy operation that doesn't have async interface.